### ISSUE TITLE
linter: the regression error caused by the no-duplicates rule

### ISSUE THREAD

**Issue by shulaoda** [oxc-project/oxc#6844]
I just upgraded to oxlint 0.10.2 and this seems to have broken things for me, unless I misunderstand the rule. I have a huge number of reported breaks in my project now, like:

```
app/javascript/src/pages/Accounts/useAccountContext.tsx:1:39: Module '@apollo/client' is imported more than once in this file [Error]
app/javascript/src/pages/Accounts/useAccountContext.tsx:8:8: Modules should not be imported multiple times in the same file [Error]
```

My full import declaration for that file is:

```tsx
import { QueryResult, useQuery } from "@apollo/client";
import {
  AccountRight,
  GetAccountDocument,
  GetAccountQuery,
  GetAccountQueryVariables,
  GrantRole,
} from "@app/graphql-codegen";
import { notEmpty } from "@app/utils";
import { Context } from "react";
import { useHistory } from "react-router-dom";
```

Downgrading to 0.10.0 (but not 0.10.1) causes the errors to disappear.

_Originally posted by @cheald in https://github.com/oxc-project/oxc/issues/6748#issuecomment-2433177892_

**Comment by shulaoda** [comment]
Let me further investigate.

**Comment by pumano** [comment]
fix present in main branch but not released

**Comment by shulaoda** [comment]
> fix present in main branch but not released. Maybe @Boshen can release patch release?

yep, please wait for the release @cheald

**Comment by cheald** [comment]
Thanks for the investigation and update! Much appreciated.
